components/button/src/dropdown-button/dropdown-button.js

```diff
@@ -187,8 +206,12 @@ DropdownButton.propTypes = {
     /**
      * Callback triggered on click.
      * Called with signature `({ name: string, value: string, open: bool }, event)`
+     * Is required when using this component in controlled mode, i.e. when
      */
-    onClick: PropTypes.func,
+    onClick: requiredIf(
+        (props) => typeof props.open === 'boolean',
+        PropTypes.func
+    ),
```

Comment:
I don't think that `onClick` needs to be required when using the component in a controlled sense.

If the consumer passes in the `open` prop then the component is controlled, full stop.

Making the `onClick` function a requirement when the component is controlled just makes the API more complex.